File: diario-classe-backend/src/main/java/com/projetointegrador/diarioclasse/entity/Nota.java
package com.projetointegrador.diarioclasse.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Data
@Entity
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "tb_notas")
public class Nota {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Double valor;
    private LocalDate dataLancamento;

    @ManyToOne
    @JoinColumn(name = "aluno_id", nullable = false)
    private Aluno aluno;

    @ManyToOne
    @JoinColumn(name = "disciplina_id", nullable = false)
    private Disciplina disciplina;

    @ManyToOne
    @JoinColumn(name = "avaliacao_id", nullable = false)
    private Avaliacao avaliacao;

    public Boolean estaAprovado(Double mediaMinima) {
        if (valor == null) return false;
        return valor >= mediaMinima;
    }}

File: diario-classe-frontend/src/models/Nota.ts
export interface Nota {
  id: number;
  alunoId: number;
  alunoNome: string;
  disciplinaId: number;
  disciplinaNome?: string;
  valor: number;
  observacao?: string;
  avaliacaoId?: number;
  avaliacaoTitulo?: string;
  dataLancamento?: string;
  mediaTurma?: string;
}
